python/model_hosting_container_standards/supervisor/README.md

```diff
@@ -0,0 +1,180 @@
+# Supervisor Process Management
+
+Provides supervisord-based process management for ML frameworks with automatic recovery and container-friendly logging.
+
+## Overview
+
+This module wraps your ML framework (vLLM, TensorRT-LLM, etc.) with supervisord to provide:
+
+- **Automatic Process Monitoring**: Detects when your service crashes or exits unexpectedly
+- **Auto-Recovery**: Automatically restarts failed processes with configurable retry limits
+- **Container-Friendly**: Exits with code 1 after max retries so orchestrators (Docker, Kubernetes) can detect failures
+- **Production Ready**: Structured logging, configurable behavior, and battle-tested supervisord underneath
+
+**Use Case**: Deploy ML frameworks on SageMaker or any container platform with automatic crash recovery and proper failure signaling.
+
+## Quick Setup
+
+### 1. Install the Package
+```bash
+pip install model-hosting-container-standards
+```
+
+### 2. Extract the Entrypoint Script
+Extract the entrypoint script from the installed package:
+```bash
+# In your Dockerfile (extracts to default: /opt/aws/supervisor-entrypoint.sh)
+RUN extract-supervisor-entrypoint
+```
+
+Or specify a custom location:
+```bash
+# In your Dockerfile
+RUN extract-supervisor-entrypoint -o /usr/local/bin/supervisor-entrypoint.sh
+```
+
+### 3. Configure Launch Command and Entrypoint
+```dockerfile
+# Set your framework's launch command
+ENV LAUNCH_COMMAND="vllm serve model --host 0.0.0.0 --port 8080"
+
+# Use supervisor entrypoint (using default path)
+ENTRYPOINT ["/opt/aws/supervisor-entrypoint.sh"]
+```
+
+### Alternative: One-line Setup
+```dockerfile
+# Install and extract in one step (uses default path: /opt/aws/supervisor-entrypoint.sh)
+RUN pip install model-hosting-container-standards && extract-supervisor-entrypoint
+
+# Still need to configure your launch command and entrypoint
+ENV LAUNCH_COMMAND="vllm serve model --host 0.0.0.0 --port 8080"
+ENTRYPOINT ["/opt/aws/supervisor-entrypoint.sh"]
+```
+
+## Configuration
+
+Configure your framework using environment variables. These can be set in your Dockerfile with `ENV` or overridden at container runtime.
+
+### Default Paths
+- **Entrypoint script**: `/opt/aws/supervisor-entrypoint.sh` (extracted by `extract-supervisor-entrypoint`)
+- **Config file**: `/tmp/supervisord.conf` (generated automatically)
+
+### Required: Launch Command
+```bash
+# Set your framework's start command
+export LAUNCH_COMMAND="vllm serve model --host 0.0.0.0 --port 8080"
+# or
+export LAUNCH_COMMAND="python -m tensorrt_llm.hlapi.llm_api --host 0.0.0.0 --port 8080"
+```
+
+### Optional Settings
+```bash
+export ENGINE_AUTO_RECOVERY=true        # Auto-restart on failure (default: true)
```

Comment:
is `ENGINE_AUTO_RECOVERY` really required ? Can we not enable/disable the feature through `ENGINE_MAX_START_RETRIES` ?